### ISSUE TITLE
AI-on-the-edge-device operation without WLAN

### ISSUE THREAD

**Issue by git2fa** [jomjol/AI-on-the-edge-device#2780]
### The Feature

Referring to the unsolved [Solution with UART alternative to WLAN in operation](https://github.com/jomjol/AI-on-the-edge-device/discussions/2629) and [Start without wifi connection](https://github.com/jomjol/AI-on-the-edge-device/issues/2197) there are a couple of reasons not to use a WLAN dependent solution.

It was not possible for @karstengit to modify the firmware to use the UART to control the AI-on-the-edge-device.
**To avoid useless WLAN radiation, power consumption and overheating it would be really fine if the firmware would disable the WLAN after a certain time, when it can't establish a WLAN connection.**

So it would be perfect to add a configuration parameter that allows to set a certain time or number of retries, after that the WLAN is disabled by the device.
When the parameter is 0 (standard) the device has the current behaviour and tries endless to connect to WLAN.

If configuration work is needed, WLAN must be present and the device is simple restarted to search the connection again.
When WLAN is off, the measurements can be fetched from the debug output on the UART.

It would be fine if this could be added to the firmware.
Thank You!

**Comment by MisterTechnik** [comment]
I think too, that a no Wi-Fi Mode would would be an amazing enhancement, because I think there would be many use cases.
For example my power meter is completely out of range of my Wi-Fi and I could only really power the device with batteries. A on-premise mode would be perfect because it would allow both, far away locations and low power consumption (because no web interface needed and deep sleep possible).
I could think of some  implementations:

- Wi-Fi Button:
    - If you want to get the data, you push a button (Boot on some models or another one added maybe) and the ESP opens an AP or connects to the configured Wi-Fi (phone hotspot) and starts the webserver so you can download a csv or something
- Wi-Fi check:
    - Sleeps x seconds, checks for configured Wi-Fi and then the above
- Manual mode:
    - Data gets saved to SD card and you can just pull the Data off that once in a while

**Comment by AAPohl** [comment]
One could think of a totally wired solution. Either a device with ethernet or a connection of Power and data (gpio, rx /tx) to mqtt ethernet device.

**Comment by Asphaltsurfer** [comment]
Ich fände eine Drahtgebundene Lösung (Lan mit POE) am besten.

**Comment by git2fa** [comment]
There are a lot of other applications possible, when this project could be used as a standalone optical reading device with low power consumption and a simple UART interface.
But it seems that an ESP application that does not use permanent WLAN is unthinkable on this planet. :-)
In the same way people don't think about the health effects of WiFi.
A wired solution can work much more reliable, when you are not nearby to solve problems with blocked channels or bad signal quality.
So why is wireless so attractive and a must have?

**Comment by SybexX** [comment]
But it's also about the available interfaces, for a lan module such as. w5500 you need SPI and all are already occupied (PSRAM + camera + SD card).
In addition, not everyone wants to tear up the basement to lay the LAN cable.

**Comment by git2fa** [comment]
> But it's also about the available interfaces, for a lan module such as. w5500 you need SPI and all are already occupied (PSRAM + camera + SD card).

There is not so plenty of data that you need an ethernet Interface.
To solve the problem with the wire distance you can simply use two RS-422-A / RS-485 interface chips like Max485 or SN75176A.
https://www.ti.com/lit/an/snla044b/snla044b.pdf

![Screenshot_20240306_123247](https://github.com/jomjol/AI-on-the-edge-device/assets/148705772/3d89c3ac-bd97-4f1d-9348-c3a333d634ac)

**Comment by SybexX** [comment]
yes, if you just want the value, if you want the website to have convenient settings..., some data will be sent because a lot of the end device is doing to relieve the load on the ESP.

**Comment by git2fa** [comment]
What's about implementing of this feature after two years?
It would be still great to enable an possible operation only with the UART without WLAN.

**Comment by tiehfood** [comment]
I'm also interested on using a ethernet device over WLAN :)

**Comment by git2fa** [comment]
> I'm also interested on using a ethernet device over WLAN :)

Up to now the AI-on-the-edge-device is used and controlled over WLAN.
This issue wants that the **data can be received over the UART** and WLAN is only used for the configuration and should be turned off afterwards.

The use of an ethernet breakout module will be difficult, because there are not enough pins to connect it.

**Comment by tiehfood** [comment]
So this is a hard hardware limitation?
Unfortunately then I cannot use this project at my location. Otherwise I would have to install a switch and an AP and do some PoE Power Splitting. I don't think that's worth it for me.
An ethernet connection would have been the optimal solution.

**Comment by SybexX** [comment]
>So this is a hard hardware limitation?

A normal ESP32 cam does not have enough free pins (you cannot use the available free pins for a W5500).
Therefore you can only use an ESP32-S3 board if you want ethernet, but this is not supported by the official firmware.

**Comment by tiehfood** [comment]
I have an ESP32-S3.
Camera works fine with esphome for example.

**Comment by SybexX** [comment]
for ESP32-S3 + esphome see:
https://github.com/jomjol/AI-on-the-edge-device/discussions/3930
https://github.com/jomjol/AI-on-the-edge-device/discussions/2963#discussioncomment-14978451

**Comment by git2fa** [comment]
> for ESP32-S3 + esphome see:
> [#3930](https://github.com/jomjol/AI-on-the-edge-device/discussions/3930)
> [#2963 (comment)](https://github.com/jomjol/AI-on-the-edge-device/discussions/2963#discussioncomment-14978451)

Will this module in connection with ESPhome enable to use a meter reading with output on the UART without WLAN?
This [UART description](https://esphome.io/components/uart/) does not say what is possible.
Maybe ESPhome will be a better solution for the own case / operation?

**Comment by SybexX** [comment]
I'm not familiar with esphome and therefore don't know what is and isn't possible with it. Operating the ESP32 without WiFi is no problem if you adapt the official firmware for it, but no one has had the time or desire to implement it yet.

**Comment by git2fa** [comment]
> I'm not familiar with esphome

It's the same for me.
But searching around there seems to be [the same problem](https://community.home-assistant.io/t/turn-off-wifi/176741).
It seems no one wants to have an ESP without WLAN, or WLAN that can be disabled / switched off. ;-)

**Comment by Slider0007** [comment]
#3874 --> Can be operated in e.g. AP standalone mode (always on or switched off after configurable delay) as well as using ethernet interface.
For UART, no special output is forseen - result needs to be parsed out of regular stream.
Alternativly using timed-off standalone AP mode: Switch on AP with a defined GPIO and download the data from time to time using WebUI.

**Comment by git2fa** [comment]
> AP standalone mode (always on or switched off after configurable delay) as well as using ethernet interface.
> For UART, no special output is forseen - result needs to be parsed out of regular stream.

Thank you - that sounds like the wanted solution.
Parsing for the result will be not the problem.

**Comment by moritzthecat** [comment]
fully support this request. May I propose alternatives dedicated for long cables

- 1-wire 100m cable length, low power and with right interface multi-device support. wide software suport.
- RS485 requires interfacce but suprt robust for long cable length

My preference would be 1-wire as I already have a local network of Sensors 
I would suggest to have a button for WiFi toggle. Wifi is certainly best setup and maintenance given current implementation.
